addon/components/pix-input-code.js

```diff
@@ -87,16 +106,25 @@ export default class PixInputCode extends Component {
   handlePaste(index, event) {
     event.preventDefault();
     event.stopPropagation();
+
     const pastedText = (event.clipboardData || window.clipboardData).getData('text');
-    [...pastedText].forEach(char => {
-      const elem = document.getElementById(`code-input-${index}`);
-      if (elem) {
-        this.focusElem(index);
-        elem.value = char;
+    const pastedTextWithOnlyValidCharacters = _removeNotAllowedCharaters(pastedText);
+
+    pastedTextWithOnlyValidCharacters.forEach(char => {
+      const input = document.getElementById(`code-input-${index}`);
+      if (input) {
+        this.focusElement(index);
+        input.value = char;
+        input.classList.add("filled");
         index++;
       }
     });
-    this.focusElem(index);
+    this.focusElement(index);
     this.triggerAction();
   }
 }
+
+function _removeNotAllowedCharaters(pastedText) {
```

Comment:
chipotage: perso je pense c'est mieux de garder un nom de methode sans negation
```suggestion
function _extractValidCharacters(pastedText) {
```